zerver/lib/user_groups.py

```diff
@@ -94,3 +94,28 @@ def get_memberships_of_users(user_group: UserGroup, members: List[UserProfile])
             user_group=user_group, user_profile__in=members
         ).values_list("user_profile_id", flat=True)
     )
+
+
+def create_system_user_groups_for_realm(realm: Realm) -> None:
+    role_system_groups_dict: Dict[int, UserGroup] = {}
+    for role in UserGroup.SYSTEM_USER_GROUP_ROLE_MAP.keys():
+        user_group = UserGroup(
+            name=UserGroup.SYSTEM_USER_GROUP_ROLE_MAP[role], realm=realm, is_system_group=True
+        )
+        role_system_groups_dict[role] = user_group
+
+    full_members_system_group = UserGroup(
+        name="Full members system group", realm=realm, is_system_group=True
+    )
+
+    role_system_groups_list = list(role_system_groups_dict.values())
+    UserGroup.objects.bulk_create(role_system_groups_list + [full_members_system_group])
+
+    for idx, user_group in enumerate(role_system_groups_list[1:]):
+        user_group.group_members.set(role_system_groups_list[: idx + 1])
+
+    members_system_group = role_system_groups_dict[UserProfile.ROLE_MEMBER]
+    guest_system_group = role_system_groups_dict[UserProfile.ROLE_GUEST]
+
+    members_system_group.group_members.add(full_members_system_group)
+    guest_system_group.group_members.add(full_members_system_group)
```

Comment:
Not sure if this approach using loop on the group lists is fine for readability or we should just add groups separetely in `group_members`